.github/workflows/build_and_test.yml

```diff
@@ -120,6 +120,56 @@ jobs:
       - name: Run mypy to check types
         run: poetry run mypy .
 
+  # Check that the vendor/postgres-* submodules point to the
+  # corresponding REL_*_STABLE_neon branches.
+  check-submodules:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Check vendor/postgres-v14 submodule reference
```

Comment:
Got an idea of an alternative approach while looking at some other PR touching submodules.

I think we would all benefit from some `make update-submodules` target that automatically bumps all submodules to the corresponding `REL_NN_STABLE_neon` branches AND update `revisions.json`

Then this CI job will be just

1. `make update-submodules`
2. Check that `git diff` is empty

without any third-party actions

Reply:
> Check that git diff is empty

Won't this fail all PRs once `REL_NN_STABLE_neon` get new commits?

Reply:
Fair, we could fix this by only running the check if submodule refs changed.

I wonder how this will work with this third-party action

Reply:
> we could fix this by only running the check if submodule refs changed

I like it!

> I wonder how this will work with this third-party action

Will we need the third-party then?

Reply:
> Will we need the third-party then?

No, I mean will it also fail in all PRs if `REL_NN_STABLE_neon` branches will change, or it's more smart